### PR TITLE
Show an X when images can't load.

### LINKED ISSUE #125: Using a resource not listed in flutter.yaml should show a red X
<a href="https://github.com/eseidelGoogle"></img></a> **Issue by [eseidelGoogle](https://github.com/eseidelGoogle)**
_Tuesday Aug 11, 2015 at 21:01 GMT_
_Originally opened as https://github.com/flutter/engine/issues/565_

---

We need to make it a visual as well as console error to use a resource not listed in that file or skyx packing won't work correctly.

Probably a change to ResourceBundle?  @abarth

### PR DESCRIPTION
**Pull request by Hixie**
When an image can't be shown, in debug mode, this shows a red X with an error message. Should make it a bit easier to debug missing images.

Fixes https://github.com/flutter/flutter/issues/125.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 3 untriaged digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/74972

**Comment by skia-gold** [comment]
Gold has detected about 2 untriaged digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/74972

**Comment by Hixie** [comment]
cc @goderbauer

File: packages/flutter/lib/src/foundation/constants.dart [hunk 1]
```diff
@@ -30,6 +44,20 @@ const bool kProfileMode = bool.fromEnvironment('dart.vm.profile', defaultValue:
 /// Since this is a const value, it can be used to indicate to the compiler that
 /// a particular block of code will not be executed in debug mode, and hence
 /// can be removed.
+///
+/// An alternative strategy is to use asserts, as in:
+///
+/// ```dart
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/constants.dart:50]: Turn this into a {@tool snippet} so it gets analyzed?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/foundation/constants.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/constants.dart:50]: done

**Comment by Hixie** [comment]
Thanks for the review, will land on green.

File: packages/flutter/lib/src/foundation/constants.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/constants.dart:51]: had to revert this, the snippet analysis logic doesn't like bare asserts.

**Pull request merged**